packages/realm-network-transport/src/types.ts

```diff
@@ -52,3 +52,202 @@ export interface NetworkTransport {
         handler: ResponseHandler,
     ): void;
 }
+
+// AbortController
+
+type AbortSignal = any;
+
+/** A controller object that allows you to abort one or more DOM requests as and when desired. */
+
+export type AbortController = {
+    /**
+     * The prototype of an instance is the class.
+     */
+    prototype: AbortController;
+
+    /**
+     * Constructs an AbortController.
+     */
+    new (): AbortController;
+
+    /**
+     * Returns the AbortSignal object associated with this object.
+     */
+    readonly signal: AbortSignal;
+    /**
+     * Invoking this method will set this object's AbortSignal's aborted flag and signal to any observers that the associated activity is to be aborted.
+     */
+    abort(): void;
+};
+
+// Environment independent Fetch API
+
+type FetchRequestInfo = FetchRequest | string;
+type FetchBodyInit = unknown;
+type FetchHeadersInit = FetchHeaders | string[][] | Record<string, string>;
+type FetchRequestCache = unknown;
+type FetchRequestCredentials = unknown;
+type FetchRequestMode = unknown;
+type FetchRequestRedirect = unknown;
+type FetchReferrerPolicy = unknown;
+type FetchRequestDestination = unknown;
+
+interface FetchBody {
+    readonly body: unknown | null;
```

Comment:
This needs to be the correct ReadableStream type for `watch()`

Reply:
I suggest we'll leave this "unknown" here and allow the consumer of this provide [an instanceof type guard](https://www.typescriptlang.org/docs/handbook/advanced-types.html#instanceof-type-guards) when consuming this API.